open-backend/controllers/message_controller.js

```diff
@@ -28,7 +28,7 @@ exports.message_write_post = async function(req, res, next) {
         content: req.body.content,
         available_at: getAvailableDate(),
         sender_id: req.user._id,
-        recipients: await getRecipients(req.user.id)
+        recipients: await getRecipients(req.user._id)
```

Comment:
The underscore is left out intentionally here. The reason is that user._id is an object, while user.id is a string. This parameter is later being compared with the results of User.aggregate (which generates random users to send the message to), but .aggregate returns "normal" objects, not mongoose documents. So the ._id property of these objects are strings, not objects. We need to ensure that we don't send messages to ourselves, so we have the condition recipient._id != req.user.id. If we used user._id instead, this statement would always be true, since the object is never equal to the string